### ISSUE TITLE
[Bug]: defaultProps Warnings

### ISSUE THREAD

**Issue by morganney** [storybookjs/storybook#27027]
### Describe the bug

Not strictly a bug, but React versions >= 18.3.0 report console warnings of unsupported features in React 19.

```
react-dom.development.js:86 Warning: WithTooltipPure: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.
```

In my particular case it was from the tooltip component used by the docs addon: https://github.com/storybookjs/storybook/blob/1e0e8220bcdc8d0988732d2f8889cdc42284d48b/code/ui/components/src/components/tooltip/WithTooltip.tsx#L123

Any particular plans to remove these warnings in version 8 of storybook, or are you going to push a new major version that supports React 19?

### To Reproduce

Use `@storybook/react` with `@storybook/addon-docs`.

### System

```bash
morgan@machine:~/code/tts-react (develop)$ ./node_modules/.bin/storybook info

Storybook Environment Info:
(node:18511) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

  System:
    OS: Linux 6.8 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (12) x64 13th Gen Intel(R) Core(TM) i5-1335U
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 22.1.0 - ~/n/bin/node
    npm: 10.7.0 - ~/n/bin/npm <----- active
  Browsers:
    Chrome: 124.0.6367.118
```

**Comment by morganney** [comment]
Looks like you should also update your `node-fetch` version used in `@storybook/core-common` to load the transitive dep `whatwg-url` >= [9.0.0](https://github.com/jsdom/whatwg-url/blob/v9.0.0/src/url-state-machine.js) and avoid the `punycode` deprecation warning.

```
morgan@machine:~/code/tts-react (develop)$ node --trace-deprecation ./node_modules/.bin/storybook info

Storybook Environment Info:
(node:50775) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
    at node:punycode:3:9
    at BuiltinModule.compileForInternalLoader (node:internal/bootstrap/realm:398:7)
    at BuiltinModule.compileForPublicLoader (node:internal/bootstrap/realm:337:10)
    at loadBuiltinModule (node:internal/modules/helpers:104:7)
    at Module._load (node:internal/modules/cjs/loader:1075:17)
    at Module.require (node:internal/modules/cjs/loader:1310:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/home/morgan/code/tts-react/node_modules/node-fetch/node_modules/whatwg-url/lib/url-state-machine.js:2:18)
    at Module._compile (node:internal/modules/cjs/loader:1480:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1564:10)

  System:
    OS: Linux 6.8 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (12) x64 13th Gen Intel(R) Core(TM) i5-1335U
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 22.1.0 - ~/n/bin/node
    npm: 10.7.0 - ~/n/bin/npm <----- active
  Browsers:
    Chrome: 124.0.6367.118

morgan@machine:~/code/tts-react (develop)$ npm ls node-fetch
tts-react-monorepo@0.0.0 /home/morgan/code/tts-react
└─┬ story@1.0.0 -> ./packages/story
  └─┬ @storybook/react-webpack5@8.0.10
    └─┬ @storybook/builder-webpack5@8.0.10
      └─┬ @storybook/core-common@8.0.10
        └── node-fetch@2.7.0

morgan@machine:~/code/tts-react (develop)$ npm ls whatwg-url
tts-react-monorepo@0.0.0 /home/morgan/code/tts-react
├─┬ jest-environment-jsdom@29.5.0
│ └─┬ jsdom@20.0.3
│   ├─┬ data-urls@3.0.2
│   │ └── whatwg-url@11.0.0 deduped
│   └── whatwg-url@11.0.0
└─┬ story@1.0.0 -> ./packages/story
  └─┬ @storybook/react-webpack5@8.0.10
    └─┬ @storybook/builder-webpack5@8.0.10
      └─┬ @storybook/core-common@8.0.10
        └─┬ node-fetch@2.7.0
          └── whatwg-url@5.0.0
```

**Comment by RBilly** [comment]
I would like to add that an effort has been made to remove usage of defaultProps in some blocks like Stories and Link (see https://github.com/storybookjs/storybook/issues/24305). 
But I still have the issue with the Source block and there may be others.

**Comment by valentinpalkovic** [comment]
Fixed by https://github.com/storybookjs/storybook/pull/26960

**Comment by morganney** [comment]
@valentinpalkovic I think there will still be the issue I mentioned with the [deprecation warning about `punycode`](https://github.com/storybookjs/storybook/issues/27027#issuecomment-2094905490).

Looks like the `next` branch [still contains an older version of `whatwg-url`](https://github.com/storybookjs/storybook/blob/next/code/yarn.lock#L21504).

**Comment by musicmixerai** [comment]
> Fixed by #26960

Any info in what release is #26960 going to be included in?

**Comment by filipbatur** [comment]
> Fixed by #26960

Any info 

> Fixed by #26960

In what release is it gonna be included?

**Comment by filipbatur** [comment]
> Fixed by #26960

In what release is it gonna be included?

**Comment by valentinpalkovic** [comment]
> @valentinpalkovic I think there will still be the issue I mentioned with the [deprecation warning about `punycode`](https://github.com/storybookjs/storybook/issues/27027#issuecomment-2094905490).
> 
> Looks like the `next` branch [still contains an older version of `whatwg-url`](https://github.com/storybookjs/storybook/blob/next/code/yarn.lock#L21504).
> 
> Should I open another ticket for that?

No, I take care of it.
